sandbox/libs/authn/src/main/java/org/opensearch/authn/internal/InternalSubject.java

```diff
@@ -64,7 +64,28 @@ public String toString() {
      */
     public void login(AuthenticationToken authenticationToken) {
         org.apache.shiro.authc.AuthenticationToken authToken = AuthenticationTokenHandler.extractShiroAuthToken(authenticationToken);
+
+        // If already authenticated, do not check login info again
+        /*
+            TODO: understand potential repercussions in following situations:
+             1. How to handle this in password change situations
+             2. Can two subjects in same environment have same principal name? if so the following check is invalid
+         */
+        if (this.isAuthenticated() && this.getPrincipal().getName().equals(authToken.getPrincipal())) {
+            return;
+        }
         // Login via shiro realm.
         shiroSubject.login(authToken);
```

Comment:
I tried replacing this line with `SecurityUtils.getSecurityManager().authenticate(authToken);` which is evaluated much faster than login. I'm not sure if the calls are interchangeable, but login calls authenticate under the hood.

See: https://github.com/apache/shiro/blob/main/core/src/main/java/org/apache/shiro/mgt/DefaultSecurityManager.java#L272

Reply:
Good find!

Reply:
I'm not sure how important createSubject call after authenticate is. From skimming through it, it seems like it sets `rememberMe` features, resolves sessions and creates/resolves principals. I can update the PR to use authenticate to skip all the other steps, but we will have to understand the impact of doing so (mainly, whether we need them for our scope)